File: Formação Lógica de Programação/JavaScript/Combinando Nomes de Pokémons.js
// Desafios JavaScript na DIO têm funções "gets" e "print" acessíveis globalmente:
// - "gets" : lê UMA linha com dado(s) de entrada (inputs) do usuário;
// - "print": imprime um texto de saída (output), pulando linha.

// Definindo a função chamada "combinandoNomesPokemons"
function combinandoNomesPokemons(palavra) {
  let palavraPokemon = palavra + "saur";
  
  return palavraPokemon;
}

// Entrada da palavra (valor simulado para teste local)
var nomeEntrada = "Bulba"; // No desafio será: gets();

var palavraGerada = combinandoNomesPokemons(nomeEntrada);

// Exibindo a palavra gerada no terminal (substituindo o print() original):
console.log(palavraGerada);

File: Formação Lógica de Programação/Python/Combinando Nomes de Pokémons.py
# Desafio Python: geralmente utilizam a função nativa "input" para leitura:
# - "input()": lê UMA linha com dado(s) de entrada do usuário;
# - "print()": imprime um texto de saída no terminal.

def combinando_nomes_pokemons(palavra):
    # Usando f-string para juntar a palavra recebida com "saur"
    palavra_pokemon = f"{palavra}saur"
    
    return palavra_pokemon

nome_entrada = "Bulba"  # No desafio será: input()

palavra_gerada = combinando_nomes_pokemons(nome_entrada)

print(palavra_gerada)